storage/grpc_client.go

```diff
@@ -1710,7 +1711,12 @@ func (w *gRPCWriter) writeObjectSpec() (*storagepb.WriteObjectSpec, error) {
 
 // read copies the data in the reader to the given buffer and reports how much
 // data was read into the buffer and if there is no more data to read (EOF).
+// Furthermore, if the attrs.ContentType is unset, the first bytes of content
+// will be sniffed for a matching content type.
 func (w *gRPCWriter) read() (int, bool, error) {
+	if w.attrs.ContentType == "" {
```

Comment:
So this happens for each call to `read()`, which happens once per chunk (if the media is larger than ChunkSize). I dug around a bit in apiary and IIUC the content sniffing happens once for the whole media input there. I'm wondering if this makes a difference and/or if content sniffing will even work when starting from the middle of a file. Any thoughts?

Perhaps we can make a test case for this, or add a check for content type to one of our tests for multi-chunk uploads.

Reply:
Once the `w.attrs.ContentType` is set (on the first `read()`), the sniffer wont be called again. The `w.attrs.ContentType` carry through each chunk. Right?